test/runtests.jl

```diff
@@ -1,128 +1,123 @@
+# UnitTest("Limiter"                                 ,"Limiters/distributed/dlimiter.jl"), # requires MPI
+UnitTest("InputOutput - hdf5"                      ,"InputOutput/hdf5.jl"),
+UnitTest("InputOutput - spectralelement2d"         ,"InputOutput/spectralelement2d.jl"),
+UnitTest("InputOutput - hybrid2dbox"               ,"InputOutput/hybrid2dbox.jl"),
+UnitTest("InputOutput - hybrid2dbox_topography"    ,"InputOutput/hybrid2dbox_topography.jl"),
+UnitTest("InputOutput - hybrid2dbox_stretched"     ,"InputOutput/hybrid2dbox_stretched.jl"),
+UnitTest("InputOutput - hybrid3dbox"               ,"InputOutput/hybrid3dbox.jl"),
+UnitTest("InputOutput - hybrid3dcubedsphere"       ,"InputOutput/hybrid3dcubedsphere.jl"),
+UnitTest("InputOutput - hybrid3dcubedsphere_topo"  ,"InputOutput/hybrid3dcubedsphere_topography.jl"),
+UnitTest("Array interpolation"                     ,"Remapping/interpolate_array.jl"),
+UnitTest("Array interpolation"                     ,"Remapping/distributed_remapping.jl"),
+UnitTest("Aqua"                                    ,"aqua.jl"),
+UnitTest("GPU - cuda"                              ,"gpu/cuda.jl";meta=:gpu_only),
+UnitTest("GPU - data"                              ,"DataLayouts/cuda.jl";meta=:gpu_only),
+UnitTest("Spaces - serial CUDA DSS on CubedSphere" ,"Spaces/ddss1_cs.jl";meta=:gpu_only),
+UnitTest("Operators - spectral element CUDA"       ,"Operators/spectralelement/rectilinear_cuda.jl";meta=:gpu_only),
+UnitTest("Operators - finite difference CUDA"      ,"Operators/hybrid/unit_cuda.jl";meta=:gpu_only),
+UnitTest("Operators - extruded sphere space ops"   ,"Operators/hybrid/extruded_sphere_cuda.jl";meta=:gpu_only),
+UnitTest("Operators - extruded 3dbox space ops"    ,"Operators/hybrid/extruded_3dbox_cuda.jl";meta=:gpu_only),
+UnitTest("Fields - CUDA mapreduce"                 ,"Fields/reduction_cuda.jl";meta=:gpu_only),
+]
+#! format: on
 
-    @safetestset "Hypsography - 2d" begin @time include("Hypsography/2d.jl") end
-    @safetestset "Hypsography - 3d sphere" begin @time include("Hypsography/3dsphere.jl") end
+# `validate_tests` returns one of (`:duplicate_file`, `:non_existent_file`, `:pass`)
+err = validate_tests(unit_tests; test_path = @__DIR__)
 
-    @safetestset "Remapping" begin @time include("Operators/remapping.jl") end
+import ClimaComms
+ClimaComms.@import_required_backends
 
-    # now part of buildkite
-    # @safetestset "Limiter" begin @time include("Limiters/limiter.jl") end
-    # @safetestset "Distributed limiters" begin @time include("Limiters/distributed.jl") end
+filter!(
+    test -> !(
+        test.meta == :gpu_only &&
+        !(ClimaComms.device() isa ClimaComms.CUDADevice)
+    ),
+    unit_tests,
+)
+# Note: for `fail_fast = false`, the tests are all wrapped in `@testset "Unit tests"`
+#       so output is suppressed until all tests are complete.
+fail_fast = true
+prevent_leaky_tests = !isnothing(get(ENV, "CI", nothing)) # prevent leaky tests on CI)
```

Comment:
I am not a big fan of using env variables like this, especially when they are so generic. On people's shells, `CI` could be defined for reasons that are unrelated to ClimaCore (in the same way `HOME`, `EDITOR`, `TERM` and several other variables are typicallly defined). The value is also not checked, only its existance. I don't think tests should silently beheve differently depending on this. 

If you want to use env variable, I suggest you change to it `CLIMACORE_TESTS_PREVENT_LEAKY_TESTS` and define this in the GitHub runner by adding it to the workflow step.

Reply:
Yeah, I tend to prefer more specific ones also, but I figured that this was relatively standard (and somewhat easy/portable) since GHA and buildkite already define `ENV["CI"]` 🤷🏻.

Reply:
I'll merge for now, but I'm happy to update it if anyone runs into problems.